lib/puppet/face/config.rb

```diff
@@ -56,6 +59,13 @@
 
     when_invoked do |*args|
       options = args.pop
+
+      @default_section = false
+      if (options[:section] == DEFAULT_SECTION_MARKER)
```

Comment:
nit, can you drop the parentheses, it's not so common ruby unless it's needed for operator precedence. Also there are a few typos in the commit message.